hpedockerplugin/hpe_storage_api.py

```diff
@@ -1071,3 +1076,50 @@ def volumedriver_list(self, body):
 
         response = json.dumps({u"Err": '', u"Volumes": volumelist})
         return response
+
+    def revert_to_snapshot(self, name, opts=None):
+        
+        contents = json.loads(name.content.getvalue())
+        if 'Name' not in contents:
+            msg = (_('revert snapshot failed, error is : Name is required'))
+            LOG.errpr(msg)
+            raise exception.HPEPluginCreateException(reason=msg)
+        snapname = contents['Name']
+        volumename = str(contents['Opts']['promote'])
+        lock_aquired = False
+        try:
+            self._etcd.try_lock_volname(volumename)
+            lock_aquired = True
+
+            volume = self._etcd.get_vol_byname(volumename)
+            if volume is None:
+                msg = (_LE('Volume: %s does not exist' % volumename))
+                LOG.info(msg)
+                response = json.dumps({u"Err": msg})
+                return response
+
+            snapshots = volume['snapshots']
+            LOG.info("Getting snapshot by name: %s" % snapname)
+            snapshot, idx = self._get_snapshot_by_name(snapshots, snapname)
+            if snapshot:
+                LOG.info("Found snapshot by name %s" % snapname)
+                self.hpeplugin_driver.revert_snap_to_vol(volume, snapshot)
+                response = json.dumps({u"Err": ''})
+                return response
+            else:
+                msg = (_LE('snapshot: %s does not exist!' % snapname))
+                LOG.info(msg)
+                response = json.dumps({u"Err": msg})
+                return response
+        except Exception:

```

Comment:
Catch PromoteSnapshotException here so that appropriate message goes into log

Reply:
okay